src/include/type/type_id.h

```diff
@@ -22,10 +16,7 @@ enum class TypeId : uint8_t {
   DECIMAL,
   TIMESTAMP,
   DATE,
-  STRING,
   VARCHAR,
-  VARBINARY,
-  ARRAY,
```

Comment:
We should talk about what it will take to add back ARRAY after we merge this PR.